sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -244,7 +244,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def json(path: String): DataFrame = {
     // This method ensures that calls that explicit need single argument works, see SPARK-16009
-    json(Seq(path): _*)
+    option("path", path).json(Seq.empty: _*)
```

Comment:
nvm. I found the issue. Json, parquet and the other file formats will not affect the data source API developers. : ) Let me revert the changes.